#include "vulnerabilityeditorwidget.h"
#include "ui_vulnerabilityeditorwidget.h"
#include <cstdlib>

using namespace Lonely;

VulnerabilityEditorWidget::VulnerabilityEditorWidget(QWidget *parent) :
    QWidget(parent),
    settings_(NULL),
    ui(new Ui::VulnerabilityEditorWidget)
{
    ui->setupUi(this);
}

VulnerabilityEditorWidget::~VulnerabilityEditorWidget()
{
    delete ui;
}

void VulnerabilityEditorWidget::editSettings(LaylaVulnerabilitySettings& settings__) {
    settings_ = &settings__;

    ui->pistolBox->setChecked(settings_->pistol);
    ui->machineGunBox->setChecked(settings_->machineGun);
    ui->grenadeBox->setChecked(settings_->grenade);
    ui->flamethrowerBox->setChecked(settings_->flamethrower);
    ui->axeBox->setChecked(settings_->axe);
    ui->bazookaBox->setChecked(settings_->bazooka);
    ui->saberBox->setChecked(settings_->saber);
    ui->shieldBox->setChecked(settings_->shield);
}

bool VulnerabilityEditorWidget::settingsLoaded() const {
    return (settings_ != NULL);
}

void VulnerabilityEditorWidget::on_pistolBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->pistol = checked;
    }
}

void VulnerabilityEditorWidget::on_machineGunBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->machineGun = checked;
    }
}

void VulnerabilityEditorWidget::on_grenadeBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->grenade = checked;
    }
}

void VulnerabilityEditorWidget::on_flamethrowerBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->flamethrower = checked;
    }
}

void VulnerabilityEditorWidget::on_axeBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->axe = checked;
    }
}

void VulnerabilityEditorWidget::on_bazookaBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->bazooka = checked;
    }
}

void VulnerabilityEditorWidget::on_saberBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->saber = checked;
    }
}

void VulnerabilityEditorWidget::on_shieldBox_clicked(bool checked)
{
    if (settings_ != NULL) {
        settings_->shield = checked;
    }
}
